
# Transcript Quantification

The pipeline uses RSEM (RNA-Seq by Expectation Maximization) to quantify gene and isoform abundances from RNA-seq data. It accepts alignments to the reference transcriptome generated by STAR.

## Quantifying Transcript Abundance

##### Transcript abundance quantification

```text
rsem-calculate-expression \
    --fragment-length-max 1000 \
    --no-bam-output \
    --paired-end \
    --estimate-rspd \
    --calc-ci \
    --forward-prob 0.0 \
    --bam transcriptome.bam \
    rsem_reference \
    OUT
```

*Note: The above command assumes a paired-end, RF/fr-firststrand stranded protocol. The RSEM reference was generated using a slightly modified version of the reference genome that excludes ALT, HLA and Decoy contigs. Please refer to the reference files section for further details.*

## Implementation with RSEM

The pipeline implements [RSEM](https://github.com/deweylab/RSEM) version v1.3.3. The software call is executed through a Python wrapper.

## Source Code

All the relevant code can be accessed in the GitHub repository:

  - [run_RSEM.py](https://github.com/smaht-dac/rnaseq-pipelines/blob/main/dockerfiles/gtex_v10/src/run_RSEM.py)<sup><sub>1</sub></sup> [RSEM]

<sub><b>1</b>: *Original author: Francois Aguet*</sub>

---

<!-- This section relies on the html links generated by GitHub Pages 
and will not render correctly in Markdown -->
<div style="text-align: right">
    <a href="/pipelines-docs/"> Home </a> -
    <a href="0_Overview.html"> Overview </a> -
    <a href="1_Alignment.html"> Alignment </a> -
    <a href="2_Duplicate_Reads.html"> Duplicate Reads </a> -
    <a> <b>  Transcripts </b> </a> -
    <a href="4_Gene_Quantification.html"> Genes </a>
</div>